Первая версия программы

elena-zavadskaya opened the pull request.

elena-zavadskaya committed:
First commit

elena-zavadskaya committed:
Точка с запятой

elena-zavadskaya committed:
Добавила условия на проверку корректности ввода.

kgornostaeva commented on src/main/java/Main.java:
```diff
@@ -1,6 +1,68 @@
+        requestProductInformation(calculator, numberOfPeople);
+    }
+
+    public static int splitTheBill() {
```
🍏 Этот и следующий методы также можно вынести в отдельный класс, который будет интерфейсом общения с пользователем

kgornostaeva commented on src/main/java/Main.java:
```diff
@@ -1,6 +1,68 @@
+            System.out.println("В этом случае нет смысла ничего считать и делить. Давайте попробуем снова.");
+            splitTheBill();
+        } else if (numberOfPeople < 1) {
+            System.out.println("Это некорректное значение для подсчёта. Давайте попробуем снова.");
```
🍏 Классно было бы подсказать пользователю, какой ввод от него ожидается

kgornostaeva commented on src/main/java/Main.java:
```diff
@@ -1,6 +1,68 @@
+        int numberOfPeople = 0;
+        System.out.println("На скольких человек необходимо разделить счёт?");
+
+        if (scanner.hasNextInt()) {
```
👏

kgornostaeva commented on src/main/java/Main.java:
```diff
@@ -1,6 +1,68 @@
+            requestProductInformation(calculator, numberOfPeople);
+        }
+
+        System.out.println("Введите стоимость товара.");
```
🍏 Круто было бы показать пользователю формат ввода. Например, [руб,коп]

kgornostaeva commented on src/main/java/Main.java:
```diff
@@ -1,6 +1,68 @@
+        calculator.count(numberOfPeople);
+
+        System.out.println("Товар успешно добавлен.");
+        System.out.println("Хотите добавить еще один товар?");
```
🍏 А здесь было бы хорошо написать, как пользователь может выйти из сценария

elena-zavadskaya committed:
Добавила условия на проверку корректности ввода.
Исправила выход из программы.
Вынесла методы для взаимодействия с пользователем в отдельный класс.
Добавила пояснения к формату ввода.

elena-zavadskaya committed:
Добавила условия на проверку корректности ввода.
Исправила выход из программы.
Вынесла методы для взаимодействия с пользователем в отдельный класс.
Добавила пояснения к формату ввода.

kgornostaeva commented on src/main/java/Calculator.java:
```diff
@@ -0,0 +1,43 @@
+        int accountFormat1 = (int) account % 100;
+        int accountFormat2 = (int) account % 10;
+
+        if (accountFormat1 >= 2 && accountFormat1 <= 4) {
```
🍏 Этот форматтер тоже можно вынести в отдельную функцию, чтобы при желании его можно было использовать и в других сценариях)

elena-zavadskaya committed:
Заменила рекурсию на циклы.

elena-zavadskaya committed:
Вынесла обработку слова "рубль" в отдельный метод.
